### PR TITLE
Add a cookie to the token response

### PR DESCRIPTION
**Pull request by drsnyder**
In this PR, I add a cookie to the token response and add functional tests for the cookie. This is to satisfy [SERVICES-290](https://wikia-inc.atlassian.net/browse/SERVICES-290).

/cc @Wikia/services-team 

TODO:
 * [x] Parse the expires date and confirm that it's :older_man:

### CODE REVIEW

File: helios/oauth_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package helios
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+  "time"
```
Comment (aquilax) [inline helios/oauth_controller_test.go]: Formatting is off here :european_post_office:

Comment (drsnyder) [inline helios/oauth_controller_test.go]: ty :). fixed in b207aff

File: helios/oauth_controller.go [hunk 2]
```diff
@@ -10,6 +10,11 @@ import (
 	"github.com/Wikia/helios/utils"
 	"github.com/influxdb/influxdb/client"
 	"net/http"
+	"time"
+)
+
+const (
+	AccessToken = "access_token"
```
Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: We could name it AccessTokenKey to be more descriptive.

File: helios/oauth_controller.go [hunk 3]
```diff
@@ -155,3 +162,17 @@ func (controller *OAuthController) setOutputError(resp *osin.Response, errorCode
...
+
+func addAccessTokenToCookie(resp *osin.Response, responseWriter http.ResponseWriter) {
+	access_token, exists := resp.Output["access_token"]
+	if exists {
+		cookie := createAccessTokenCookie(access_token.(string))
+		http.SetCookie(responseWriter, cookie)
+	}
+}
+
+
```
Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: Redundant line :panda_face:

File: helios/oauth_controller.go [hunk 4]
```diff
@@ -155,3 +162,17 @@ func (controller *OAuthController) setOutputError(resp *osin.Response, errorCode
...
+func createAccessTokenCookie(access_token string) (*http.Cookie) {
```
Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: use camel case here access_token -> accessToken :camel:

Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: I don't see any advantage here from returning the pointer. We could return regular struct : ```return http.Cookie{Name: AccessToken, Value: access_token, Expires: cookieExpireDate}``` and then use pointer like: ```http.SetCookie(responseWriter, &cookie)```. :cookie:

Comment (drsnyder) [inline helios/oauth_controller.go]: Both of these are addressed in 6a59b58.

**Comment by ArturKlajnerok** [comment]
With small cosmetics changes LGTM :+1:

File: helios/oauth_controller.go (hunk 2 above)
Comment (drsnyder) [inline helios/oauth_controller.go]: Fixed in 535b4e5

File: helios/oauth_controller.go (hunk 4 above)
Comment (drsnyder) [inline helios/oauth_controller.go]: Ok, changed the return value in 4ac886f.

File: helios/oauth_controller.go [hunk 5]
```diff
@@ -155,3 +162,16 @@ func (controller *OAuthController) setOutputError(resp *osin.Response, errorCode
...
+	accessToken, exists := resp.Output["access_token"]
+	if exists {
+		cookie := createAccessTokenCookie(accessToken.(string))
+		http.SetCookie(responseWriter, &cookie)
+	}
+}
+
+func createAccessTokenCookie(accessToken string) (http.Cookie) {
+	cookieExpireDate := time.Now().AddDate(10, 0, 0) // 10 years
+	return http.Cookie{Name: AccessTokenKey, Value: accessToken, Expires: cookieExpireDate}
```
Comment (Grunny) [inline helios/oauth_controller.go]: @drsnyder Since this is effectively a session cookie, can we set HttpOnly to true on it?

Comment (drsnyder) [inline helios/oauth_controller.go]: Good point. My general sense is that the cookie is incomplete. For example, we also aren't specifying a domain even though it should be implicit. I'll create a separate ticket for cleaning it up.

**Pull request merged**